server/src/item.rs

```diff
@@ -32,8 +32,32 @@ impl Item {
     }
 
     #[zbus(out_args("secret"))]
-    pub async fn get_secret(&self, _session: ObjectPath<'_>) -> Result<SecretInner, ServiceError> {
-        todo!()
+    pub async fn get_secret(
+        &self,
+        session: OwnedObjectPath,
+    ) -> Result<(SecretInner,), ServiceError> {
+        // wip
+        let inner = self.inner.lock().await;
+        let manager = self.manager.lock().await;
+        let secret = inner.secret();
+        let session = manager.session(&session);
+        if session.is_none() {
+            tracing::error!("The session does not exist");
+            return Err(ServiceError::NoSession("NoSession".to_string()));
```

Comment:
It is useless to return NoSession in an error message that is already named NoSession. Instead pass the info about which session or a more expressive error message :)

Reply:
Thanks for the review. Yeah of course. This was just placed temporarily.